File: basekitk/src/main/kotlin/com/mcgars/basekitk/features/db/ToolContentProvider.java
package com.mcgars.basekitk.features.db;

/**
 * Created by Феофилактов on 24.11.2014.
 */
public abstract class ToolContentProvider {
    public abstract android.net.Uri getContentUri(String path);

    public abstract android.net.Uri getContentUriGroupBy(String path, String groupBy);

    public abstract android.net.Uri getContentWithLimitUri(String path, int limit);

    public abstract android.net.Uri getNoNotifyContentUri(String path);

    public abstract android.net.Uri getNoNotifyContentUri(String path, long id);

    public static void setIsDebug(boolean debug){
        SQLBuilder.Companion.setIsDebug(debug);
    }

}

File: basekitk/src/main/kotlin/com/mcgars/basekitk/features/db/ToolContentProvider.kt
package com.mcgars.basekitk.features.db

/**
 * Created by Феофилактов on 24.11.2014.
 */
abstract class ToolContentProvider {
    abstract fun getContentUri(path: String): android.net.Uri

    abstract fun getContentUriGroupBy(path: String, groupBy: String): android.net.Uri

    abstract fun getContentWithLimitUri(path: String, limit: Int): android.net.Uri

    abstract fun getNoNotifyContentUri(path: String): android.net.Uri

    abstract fun getNoNotifyContentUri(path: String, id: Long): android.net.Uri

    companion object {
        fun setIsDebug(debug: Boolean) {
            SQLBuilder.isDebug = debug
        }
    }

}
